server_tests/test/clientTests.js

```diff
@@ -1102,26 +1095,20 @@ describe("RUST API TEST SUITE", function() {
             }, wrapError);
         });
 
-        it.skip("Expect success. /osc/commands/status successfully grabs command status after take picture has been called", function() {
+        it("Expect success. /osc/commands/status successfully grabs command status after take picture has been called", function() {
             this.timeout(timeoutValue);
             var deferred = Q.defer();
 
-            return Q.all(
-                [
-                    testClient.takePicture(sessionId, function(res) {
-                        var commandId = res.body.id;
-                        testClient.commandsStatus(commandId)
-
-                        .then(validate.inProgress(res.body, schema.names.commandsStatus))
-                        .then(deferred.resolve, deferred.rejecet)
+            return Q.all([testClient.takePicture(sessionId, function(res) {
+                var commandId = res.body.id;
 
-                    })
-                    .then( function onSuccess (res) {
-                        validate.done(res.body, schema.names.commandTakePicture);
-                    }, wrapError)
-                    .then(() => console.log('takePictureFinished')),
-                    deferred.promise
-                ])
+                return testClient.commandsStatus(commandId)
```

Comment:
So, an actual call to `commandsStatus` isn't necessary here. We already have the data to validate (`res`) from the status callback. Also, this function doesn't need to return a promise since the library won't do anything with it; that's why we need to use deferred in the first place.

We just need to feed whether `res` is valid to the deferred promise. This could be done with an empty promise instead of the `commandsStatus()` call:

    Q().then(() => validate.inProgress(res.body, ...)).then(deferred.resolve, deferred.reject);

otherwise something like:

    try {
        validate.inProgress(...);
        deferred.resolve();
    } catch (err) {
        deferred.reject(err);
    }